codice/ddf
# [2.19.x] G-9795 G-9816 Fix saving search forms that use the Date BETWEEN and Date EMPTY operators

#### What does this PR do?
Prior to this change, trying to save a search form that used a Date BETWEEN or Date EMPTY operator would fail with a red error banner on the UI. This fixes the issue. 

#### Who is reviewing it? 
#### Ask 2 committers to review/merge the PR and tag them here.
@brianfelix
@cassandrabailey293
@millerw8
@mojogitoverhere

#### How should this be tested?
Testing "Between"
- Open search forms and start a new form.
- Pick any Date attribute, such as `Created Date` or `Metadata Revision Date`, and pick the `BETWEEN` operator.
- Ensure the form saves.
- Ensure refreshing the page doesn't result in errors and the form gets resurrected correctly. 
- Repeat the process with the exact same predicate but have it embedded in a logical operator of some kind (and/or).
- Open Solr, find the form metacard, grab the XML payload in the `ui.template-filter` attribute and ensure when prettified it looks like this (plus any extra data that might be there for the logical case):
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Filter xmlns="http://www.opengis.net/fes/2.0" xmlns:ns2="http://www.opengis.net/ows/1.1" xmlns:ns3="http://www.w3.org/1999/xlink">
	<During>
		<ValueReference>created</ValueReference>
		<Literal>1544447380000/1544447380000</Literal>
	</During>
</Filter>
```

Testing "Empty"
- Open search forms and start a new form.
- Pick any Date attribute, such as `Created Date` or `Metadata Revision Date`, and pick the `EMPTY` operator.
- Ensure the form saves.
- Ensure refreshing the page doesn't result in errors and the form gets resurrected correctly. 
- Repeat the process with the exact same predicate but have it embedded in a logical operator of some kind (and/or).
- Open Solr, find the form metacard, grab the XML payload in the `ui.template-filter` attribute and ensure when prettified it looks like this (plus any extra data that might be there for the logical case):
```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Filter xmlns="http://www.opengis.net/fes/2.0" xmlns:ns2="http://www.opengis.net/ows/1.1" xmlns:ns3="http://www.w3.org/1999/xlink">
	<PropertyIsNil>
		<ValueReference>created</ValueReference>
	</PropertyIsNil>
</Filter>
```

To facilitate the above tests, here is an additional example when more predicates are added into the mix. 
```
<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>
<Filter xmlns=\"http://www.opengis.net/fes/2.0\" xmlns:ns2=\"http://www.opengis.net/ows/1.1\" xmlns:ns3=\"http://www.w3.org/1999/xlink\">
	<And>
		<PropertyIsLike wildCard=\"%\" singleChar=\"_\" escapeChar=\"\\\">
			<ValueReference>anyText</ValueReference>
			<Literal>invest*igation</Literal>
		</PropertyIsLike>
		<Or>
			<PropertyIsNil>
				<ValueReference>created</ValueReference>
			</PropertyIsNil>
			<During>
				<ValueReference>created</ValueReference>
				<Literal>1613692800000/1613779200000</Literal>
			</During>
		</Or>
	</And>
</Filter>
```

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
